### PR TITLE
Create listfiles.md

### PR DESCRIPTION
**Pull request by EpixScripts**

### CODE REVIEW

File: api/filesystem/listfiles.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# listfiles
+**Signature:** `listfiles(folderPath: string): {string}` <br>
```
Comment (LuaOranges) [inline api/filesystem/listfiles.md:2]: The return type should be: {[number]: string}

Comment (EpixScripts) [reply api/filesystem/listfiles.md:2]: {string} is actually valid shorthand syntax in Luau for {[number]: string}, you can check the docs if you like.

Comment (EpixScripts) [reply api/filesystem/listfiles.md:2]: See https://luau-lang.org/typecheck#table-indexers

Comment (LuaOranges) [reply api/filesystem/listfiles.md:2]: I can't believe I wrote all that code without knowing this!

**Comment by ShowerHead-FluxTeam** [comment]
Looks good to me

**Comment by icebearc** [comment]
looks good

**Comment by LuaOranges** [comment]
Majority of UNC council has approved, merging.

**Pull request merged**